Visually appealing Security Level change

DrDiasyl opened the pull request:
## About The Pull Request

During the code Red or above special alarm lights start blazing red light to signal everyone to remind them of the stressful situation station is in.
![full test of alert](https://user-images.githubusercontent.com/42353186/209127219-bf90fc99-4ed2-430f-9757-0e401fd55e0a.gif)
![full test of delta alert](https://user-images.githubusercontent.com/42353186/209128021-911289b0-eb64-4d91-a38a-84859ea564f0.gif)
_Low quality due gif compressing, doesn't fully represent how it looks in game_



Status displays themselves and their screens now got a new flashed look with better sprites (including AI display sprites, but they just got the interlaced screen effect to fit near the status displays).
They also automatically turn on when the security level changes. Biohazard status displays automatically turn on when a Level 5 or 7 outbreak gets announced.
![screen](https://user-images.githubusercontent.com/42353186/209126349-bc04e855-c397-4f80-afa1-5bf8442e143a.png)
![green](https://user-images.githubusercontent.com/42353186/209126076-dbca2879-7c55-4ddd-bc71-27eabaa9901f.gif)
![lockdown](https://user-images.githubusercontent.com/42353186/209126079-2f5a1897-f36c-4d80-b777-9d9fd7c53896.gif)
![radiation](https://user-images.githubusercontent.com/42353186/209126081-8b74568e-b0d0-49d4-bb8b-6cba6e518d09.gif)
![red](https://user-images.githubusercontent.com/42353186/209126083-33b51f7e-a7ab-45bd-83b0-6747d52a04bf.gif)
![biohazard](https://user-images.githubusercontent.com/42353186/209126086-47d145e2-6925-4157-a183-eb81564d07c7.gif)
![blue](https://user-images.githubusercontent.com/42353186/209126088-70500133-0e1f-45a3-913f-84ce15f122e5.gif)
![default](https://user-images.githubusercontent.com/42353186/209126091-615f9260-1c11-457b-9fae-32cb9d2f5bba.gif)
![Delta](https://user-images.githubusercontent.com/42353186/209126096-729a38d5-6f8d-4862-945e-98568e053c81.gif)

Now in the command console, you cannot turn on the Red code status display, there is now an exclusive button that turns on the status display of the current security level.
![image](https://user-images.githubusercontent.com/42353186/209128442-35dd9768-76bf-4a1c-9dd4-587168fd9c79.png)

The announcement of code Red and Delta gets a more distinctive sound. Code red now uses the new "notice3.ogg" while code Delta uses "airraid.ogg" which was changed to be more sci-fi and to not get cut in the middle.

https://user-images.githubusercontent.com/42353186/209131242-36911499-af7e-4d32-81b2-d4a1c70af412.mp4

https://user-images.githubusercontent.com/42353186/209131251-86c1adb3-8259-4e7f-8c0f-c0b91421fe49.mp4


"bloblarm.ogg" now doesn't get cut off abruptly, making it better for the ears of people.

Fixed a bug where the floor light turns invisible when a fire alarm is activated in the area it's in.
## Why It's Good For The Game

Security alert level always felt like it didn't have enough intimidation on players, when code Red gets declared the best reaction you can get from players is "Why?" after which they forget about it and behave casually. With this players are now visually reminded that there is currently a bad situation on the station, this will put a psychological effect on players which will _theoretically_ make them follow the code Red guidelines and surely not forget about it. Status displays and alarm lights will do the most job for this, also with them reminding people about Level 5 and Level 7 Outbreaks.

Old status displays sprites were just really bad, they are probably extremely old with weird color palettes and needed refreshing for a long time. The display itself seemed flat and badly shaded.

Different sound effects for code Red and code Delta makes them much more distinctive from others' alert levels (as now each of them uses a different sound) which not only helps players understand what's going on faster but also can help people who can't read big red letters in chat saying what code has been declared (Especially with Red and Delta sounds being pretty intimidating)

**Draft for now because I didn't place alarm lights around all the stations, I would really appreciate if someone will help me with that and send the map files with them placed, thanks you.**
## Changelog
:cl: DrDiasyl aka DrTuxedo
add: Status displays now automatically change to the current security level one.
add: Status displays now automatically show biohazard warning after Level 5 or Level 7 outbreak gets announced.
add: New alarm lights (beacons) around the stations which turn on during code Red or Delta to remind players of the dire situation on the station.
add: Now in the command console, there is a new button "Security Level Code" which replaces the "Code Red" one in the status displays control panel, it shows the current security level display.
fix: Floor lights no longer turn invisible after having fire alarm activated in their area.
fix: "bloblarm.ogg" no longer abruptly shuts off, making it pleasing for the ears.
soundadd: New "notice3.ogg" sound which plays during code Red.
soundadd: "airraid.ogg" sound got changed for a more sci-fi better quality one.
imageadd: Status displays got new flashed sprites, making them much more pleasing and attention-grabbing to the eye.
/:cl:

DrDiasyl committed:
New notice sounds

New more aggressive "notice3" for Code Red

New "airraid" sound which is now used in delta

DrDiasyl committed:
New alarm light

The alarm light shines only during code red and higher, indicating to uninformed people that the station is in a state of emergency.

Works fine, for now, thanks to "Oranges" and "cringe" from the coderbus. Just have a little bug to fix which is it turning on if you take the light out and in, and also the fact that you can't build any new ones.

Co-Authored-By: oranges <email@oranges.net.nz>

DrDiasyl committed:
blobalarm.ogg now doesn't end abruptly

Added fade out at it's end so it doesn't end abruptly anymore.

DrDiasyl committed:
Light alarm new file, shows a span, fixed a bug

Light alarm now has its file which is defined.

Now there is a span that shows when it's examined, indicating if the station is in a state of emergency.

ALSO! I have fixed a bug that made floor lights transparent when major_emergency = true.

DrDiasyl committed:
Made alarm sturdier & lightswitch doesn't affect it

DrDiasyl committed:
Removes examine prompts

Okay, it's actually kinda niche and useless NGL

DrDiasyl committed:
Status displays now show alert level

New sprites for status displays, added new status displays for delta and green alerts. A new sprite for the frame itself.

Now every status display (excluding entertainment) has an interlaced screen effect.

Added some comments around the code.

Now instead of the "Red Alert" status display button on the comms console, there is "Security Alert Level" which automatically shows the current sec level to avoid abuse. But also now you can turn on the "radiation" screen for status displays.

Status displays automatically show a "biohazard" display when level 5 and level 7 outbreaks get announced.

DrDiasyl committed:
Merge branch 'master' into alert-codes

DrDiasyl commented:
![image](https://user-images.githubusercontent.com/42353186/209133802-0c82a645-90eb-494b-b1fe-52018a75df7a.png)

What the hell I should do with this?

Mothblocks commented:
you can fix that by not putting anything more on the dumbass radio system

Mothblocks commented on code/modules/power/lighting/light_alarm.dm:
```diff
@@ -0,0 +1,35 @@
+	. = ..()
+	RegisterSignal(SSsecurity_level, COMSIG_SECURITY_LEVEL_CHANGED, PROC_REF(emergency))
+
+/obj/machinery/light/attackby(obj/item/tool, mob/living/user, params) //If people decide to stick a new lightbulb in it, it will not turn on by itself
```
supposed to be on alarm

DrDiasyl replied:
Huh?

Mothblocks replied:
`/obj/machinery/light/attackby`, you mean `light/alarm/attackby`

DrDiasyl replied:
> `/obj/machinery/light/attackby`, you mean `light/alarm/attackby`

Oooh this, for some reason when I changed it to "alarm" I was unable to change it's lightbulb

DrDiasyl commented:
> you can fix that by not putting anything more on the dumbass radio system

Ahh, it's that stupid?

Mothblocks commented:
yeah we are trying to get rid of it completely

DrDiasyl commented:
> yeah we are trying to get rid of it completely

I can solve this by making a global function for this status screen change, can I?

Mothblocks commented:
global signal works

Serxule commented:
imo the sprites are cool but there's way too much happening (when the alerts will be happening when too much is happening), i feel they should be kept simple (also why are some of them like flashing neon signs you'd see at a club?)

DrDiasyl commented:
> imo the sprites are cool but there's way too much happening (when the alerts will be happening when too much is happening), i feel they should be kept simple to convey what is happening at a glance, (eg old red alert was "ALERT" with a red background which easily conveys that its red alert at a glance), some of them look like flashing signs you'd see at a casino

Just text is bland and tasteless.

DrDiasyl commented:
> you can fix that by not putting anything more on the dumbass radio system

Suggest something then.

Serxule commented:
just search "scifi alert sign" on google images, bam, ideas

DrDiasyl commented:
I think they are fine, most images I found are just text with some triangles in them, but this will mess with clarity as I already had a problem fitting the text in while it was clear.

DrDiasyl committed:
status_alarm is now a global proc

This fixes the overloading of other scripts due putting too much on "on the dumbass radio system"

DrDiasyl committed:
Merge branch 'alert-codes' of https://github.com/DrDiasyl/tgstation into alert-codes

DrDiasyl commented:
@Mothblocks I have no idea how to fix this, help.

DrDiasyl committed:
Merge branch 'master' into alert-codes

DrDiasyl commented on code/modules/antagonists/blob/overmind.dm:
```diff
@@ -154,6 +154,7 @@ GLOBAL_LIST_EMPTY(blob_nodes)
 
 	if(announcement_time && (world.time >= announcement_time || blobs_legit.len >= announcement_size) && !has_announced)
 		priority_announce("Confirmed outbreak of level 5 biohazard aboard [station_name()]. All personnel must contain the outbreak.", "Biohazard Alert", ANNOUNCER_OUTBREAK5)
+		status_alarm("biohazard")
```
> This should be added to the random event blob so false alarms also update the status displays

This will be easy to solve. Is there no such problem with level 7 Outbreak?

iwishforducks commented:
cool as fuck

DrDiasyl commented on code/game/machinery/computer/communications.dm:
```diff
@@ -381,7 +381,18 @@
-				post_status("alert", picture)
+				if(picture == "currentalert") // Tell me how to do this using "return" and I surely will. Using this we can make it so players cannot set Delta code display during for example code Blue
+					switch(SSsecurity_level.get_current_level_as_number())
+						if(SEC_LEVEL_DELTA)
+							post_status("alert", "deltaalert")
+						if(SEC_LEVEL_RED)
+							post_status("alert", "redalert")
+						if(SEC_LEVEL_BLUE)
+							post_status("alert", "bluealert")
+						if(SEC_LEVEL_GREEN)
+							post_status("alert", "greenalert")
```
Is it worth it?

DrDiasyl replied:
Okay I actually have a easier solution

DrDiasyl committed:
Trying to fix some unchecked checks

DrDiasyl committed:
Merge branch 'alert-codes' of https://github.com/DrDiasyl/tgstation into alert-codes

celulamp commented:
> > > snip
> > 
> > 
> > Just text is bland and tasteless.
> 
> still, there's ways to make it interesting _and_ be able to convey a message at a glance, this is way too much happening + casino sign aesthetic

it does not look like casino signs
it looks like just text that gets to the point
or icon etc telling you what is going on. helps people who join round and can see whats going on instantly by looking at them.

making them flashy will make ppl notice and pay attention

Mothblocks commented:
i like them

OrionTheFox commented:
I think the ones that spell out the word letter by letter do feel more casino than hazard; maybe just having them flash the whole word as the others do would make it feel less playful and more 'shits going down'

castawaynont commented:
too flashy, I also don't like the red/yellow color scheme, it reminds me more of a food ad than a menacing and scary alert panel. try red black or red orange, maybe

DrDiasyl commented:
> I think the ones that spell out the word letter by letter do feel more casino than hazard; maybe just having them flash the whole word as the others do would make it feel less playful and more 'shits going down'

I think I can remove letter by letter one's, this will also make this PR weight less.

DrDiasyl commented:
> too flashy, I also don't like the red/yellow color scheme, it reminds me more of a food ad than a menacing and scary alert panel. try red black or red orange, maybe

Red and yellow are one of the most noticeable colors for the humans, all warning signs use yellow color and danger signs use red. I never ever had in my mind a idea of this reminding someone of a food ad

Dorsisdwarf commented:
Not sure I like the red alert lights treading on the toes of fire alarm flashing red lights.

Dorsisdwarf commented:
> > too flashy, I also don't like the red/yellow color scheme, it reminds me more of a food ad than a menacing and scary alert panel. try red black or red orange, maybe
> 
> Red and yellow are one of the most noticeable colors for the humans, all warning signs use yellow color and danger signs use red. I never ever had in my mind a idea of this reminding someone of a food ad

I've never seen an emergency warning alert that scrolls the text with a flashing multicoloured border alternating colours like I've just won £5 on the slots either

Mothblocks commented:
@tgstation/sprite-team please advise

DrDiasyl commented:
Now that I got days off I will start fixing all these bugs and issues. I will make sprite animations smaller (remove single letter lightning up)

Mothblocks commented:
i liked it =(

Twaticus reviewed (changes requested):
the alarm light is cool, the new sounds are great, but the status displays i *really* dont like. the hotdog stand color scheme and abrupt alternating letters seem extremely childish for alerts. i dont see those designs being salvageable. it may be best to revert the status displays and just keep the sounds/new flashing light. sorry

DrDiasyl commented:
> the alarm light is cool, the new sounds are great, but the status displays i *really* dont like. the hotdog stand color scheme and abrupt alternating letters seem extremely childish for alerts. i dont see those designs being salvageable. it may be best to revert the status displays and just keep the sounds/new flashing light. sorry

Pftt, I just can make better sprites, just wait a little and you will see today...

DrDiasyl commented:
Although the only one having the "hot dog" colors is the Delta

DrDiasyl commented:
I will remove all of the unnecessary frames from the status display animations leaving only the key ones, this way it won't feel that childish anymore.
There is always an exit and there is always a solution, if there is some objective in my mind I will reach **no matter the costs.**

DrDiasyl commented:
![NEW RED](https://user-images.githubusercontent.com/42353186/210218987-1b3cf00f-0a9e-4c09-9fd4-c14226d3a777.gif)
I might do all of these in this kind of style, I also might change the color palette if this red color makes _for some reason_ word "ketchup" resonate in your head

BlueMemesauce commented:
that's a really good improvement, it looks much more serious

Mothblocks commented:
@DrDiasyl dont let this one drop king

DrDiasyl commented:
> @DrDiasyl dont let this one drop king

Will try

optimumtact commented:
Open a new PR when you are ready

Ekaterina-von-Russland commented:
It's so sad that you let this die. This PR was good.

DrDiasyl commented:
> It's so sad that you let this die. This PR was good. I hope you bring it back after the feature freeze.

S O O N™️

DrDiasyl commented:
> It's so sad that you let this die. This PR was good. I hope you bring it back after the feature freeze.

IT HAPPENED TODAY

DrDiasyl commented:
> Open a new PR when you are ready

Well did

DrDiasyl commented:
> @DrDiasyl dont let this one drop king

Actually didn't drop it!
